src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheckTest.java

```diff
@@ -230,4 +230,26 @@ public void testGetRequiredTokens() {
                 .that(matchXpathCheck.getRequiredTokens())
                 .isEmpty();
     }
+
+    @Test
+    public void testExceptionMessage() {
```

Comment:
It's a copy paste of `testEvaluationException()` with exception checked. It should be possible to test it via input like:
```
/*
MatchXpath
query = count(*) div 0

*/

package com.puppycrawl.tools.checkstyle.checks.coding.matchxpath;

public class InputWithBadQuery {
}
```

Reply:
Yes, no new pure unit testing in fake tree. 
All should be on our AST and on our Check.

Reply:
>  It should be possible to test it via input like:

Yes, but we want to get the exact message of the exception 

please guide me if i miss anything 

```
    @Test
    public void testCheckWithMatchingMethodName2() throws Exception {
        final DefaultConfiguration checkConfig = createModuleConfig(MatchXpathCheck.class);
        checkConfig.addProperty("query", "count(*) div 0");
        try {
           verify(checkConfig, getPath("InputMatchXpath5.java"));
        }
        catch (XPathException ex) {
            assertWithMessage("Exception message")
                .that(ex.getMessage())
                .isEqualTo("Evaluation of Xpath query failed: count(*) div 0");
        }
    }
```

Reply:
Did you try to catch all exceptions ? I think there should be CheckstyleException 
And message will be in cause exception.

Reply:
If you tried and it didn't not, please show all all details on how you did this and what is results

Reply:
Done please check